samples/pyarrow/append_rows_with_arrow.py

```diff
@@ -0,0 +1,172 @@
+    date_2 = datetime.date(2021, 10, 1)
+
+    datetime_1 = datetime.datetime(2016, 12, 3, 14, 11, 27, 123456)
+    datetime_2 = datetime.datetime(2017, 12, 3, 14, 11, 27, 123456)
+
+    timestamp_1 = datetime.datetime(
+        1999, 12, 31, 23, 59, 59, 999999, tzinfo=datetime.timezone.utc
+    )
+    timestamp_2 = datetime.datetime(
+        2000, 12, 31, 23, 59, 59, 999999, tzinfo=datetime.timezone.utc
+    )
+
+    # Pandas Dataframe.
+    rows = []
+    for i in range(row_num):
+        row = {
+            "bool_col": True,
+            "int64_col": i,
+            "float64_col": float(i),
+            "numeric_col": decimal.Decimal("0.000000001"),
+            "bignumeric_col": decimal.Decimal("0.1234567891"),
+            "string_col": "data as string",
+            "bytes_col": str.encode("data in bytes"),
+            "date_col": datetime.date(2019, 5, 10),
+            "datetime_col": datetime_1,
+            "time_col": datetime.time(23, 59, 59, 999999),
+            "timestamp_col": timestamp_1,
+            "geography_col": "POINT(-121 41)",
+            "range_date_col": {"start": date_1, "end": date_2},
+            "range_datetime_col": {"start": datetime_1, "end": datetime_2},
+            "range_timestamp_col": {"start": timestamp_1, "end": timestamp_2},
+        }
+        rows.append(row)
+    df = pd.DataFrame(rows)
+
+    # Dataframe to PyArrow Table.
+    schema = pa.schema(
+        [
+            pa.field("bool_col", pa.bool_()),
+            pa.field("int64_col", pa.int64()),
+            pa.field("float64_col", pa.float64()),
+            pa.field("numeric_col", pa.decimal128(38, scale=9)),
+            pa.field("bignumeric_col", pa.decimal256(76, scale=38)),
+            pa.field("string_col", pa.string()),
+            pa.field("bytes_col", pa.binary()),
+            pa.field("date_col", pa.date32()),
+            pa.field("datetime_col", pa.timestamp("us")),
+            pa.field("time_col", pa.time64("us")),
+            pa.field("timestamp_col", pa.timestamp("us")),
+            pa.field("geography_col", pa.string()),
+            pa.field(
+                "range_date_col",
+                pa.struct([("start", pa.date32()), ("end", pa.date32())]),
+            ),
+            pa.field(
+                "range_datetime_col",
+                pa.struct([("start", pa.timestamp("us")), ("end", pa.timestamp("us"))]),
+            ),
+            pa.field(
+                "range_timestamp_col",
+                pa.struct([("start", pa.timestamp("us")), ("end", pa.timestamp("us"))]),
+            ),
+        ]
+    )
+    table = pa.Table.from_pandas(df, schema=schema)
+
+    # Construct request.
+    request = gapic_types.AppendRowsRequest()
+    arrow_data = gapic_types.AppendRowsRequest.ArrowData()
+    arrow_data.writer_schema.serialized_schema = table.schema.serialize().to_pybytes()
+    arrow_data.rows.serialized_record_batch = (
+        table.to_batches()[0].serialize().to_pybytes()
```

Comment:
It might be good to demonstrate at least a simple splitting strategy to show how to split a large dataframe into multiple append requests.  We want to demonstrate how to create reasonably-sized requests from a potentially much larger dataframe.  For our trivial example this is less of a concern, but we want to establish some good practices here for users who adapt this sample for their own use.

Reply:
done